architecture/service_catalog/ansible_service_broker.adoc

```diff
@@ -0,0 +1,129 @@
+To opt-in during installation or upgrade, see Installation & Configuration (TODO link).
+endif::[]
+====
+
+The Ansible service broker (ASB) is an implementation of the OSB API that
+manages applications defined by xref:service-catalog-apb[_Ansible playbook bundles (APBs)_]. APBs provide a new method for defining and distributing
+container applications in {product-title}, consisting of a bundle of Ansible
+playbooks built into a container image with an Ansible runtime. APBs leverage
+Ansible to create a standard mechanism for automating complex deployments.
+
+The design of the ASB follows this basic workflow:
+
+. A user requests list of available applications from the service catalog using
+the {product-title} web console.
+. The service catalog requests the ASB for available applications.
+. The ASB communicates with a defined container registry to learn which APBs are
+available.
+. The user issues a request to provision a specific APB.
+. The provision request makes its way to the ASB, which fulfills the user's
+request by invoking the provision method on the APB.
+
+image::asb-arch.png["Ansible Service Broker Architecture"]
+
+[[service-catalog-apb]]
+== Ansible Playbook Bundles
+
+An Ansible playbook bundle (APB) is a lightweight application definition for
+defining, allowing you to leverage existing investment in Ansible roles and
+playbooks.
+
+APBs use a simple directory with named playbooks to perform OSB API actions,
+such as provision and bind. Metadata defined in *_apb.yml_* spec file contains a
+list of required and optional parameters for use during deployment.
+
+[[service-catalog-apb-directory]]
+=== Directory Structure
+
+The following shows an example directory structure of an APB:
+
+----
+example-apb/
+├── Dockerfile
+├── apb.yml
+└── roles/
+│   └── example-apb-openshift
+│       ├── defaults
+│       │   └── main.yml
+│       └── tasks
+│           └── main.yml
+└── playbooks/
+    └── provision.yml
+    └── deprovision.yml
+    └── bind.yml
+    └── unbind.yml
+----
+
+[[service-catalog-apb-directory]]
+=== Spec File
+
+An APB spec file (*_apb.yml_*) must be edited for your specific application. For
+example, the `etherpad-apb` spec file looks as follows:
+
+----
+name: fusor/etherpad-apb
+description: Note taking web application
+bindable: true
+async: optional
+parameters:
+  - name: hostport
+    description: The host TCP port as the external endpoint
+    type: int
+    default: 9001
+  - name: db_user
+    description: Database User
+    type: string
+----
+
+For an APB that does not have any parameters, the `parameters` field should be
+blank:
+
+----
+parameters: []
+----
+
+[[service-catalog-apb-directory]]
+=== Actions
+
+The following are the actions for an APB. At a minimum, an APB must implement the provision and deprovision actions.
+
+*_provision.yml_*::
+Playbook called to handle installing application to the cluster.
+
+*_deprovision.yml_*::
+Playbook called to handle uninstalling.
+
+*_bind.yml_*::
```

Comment:
Should we mention that the bind/unbind playbooks will not be called by default (need to update the broker config to get it to run the bind/unbind playbooks in the apb)